site/en/codelabs/telink-openthread-hardware/index.lab.md

```diff
@@ -0,0 +1,739 @@
+
+### Firmware Compilation
+
+This Codelab will build two types of OpenThread firmware:
+
+* `ot-cli-ftd`,
+* and `ot-rcp`.
+
+ The compilation methods are as follows：
+
+1. Radio Co-Processor（`ot-rcp`)
+
+     ```bash
+     $ cd ~/zephyrproject
+     $ rm -rf build_ot_coprocessor
+     $ west build -b tlsr9518adk80d -d build_ot_coprocessor zephyr/samples/net/openthread/coprocessor -- -DDTC_OVERLAY_FILE="usb.overlay" -DOVERLAY_CONFIG=overlay-rcp-usb-telink.conf
+     ```
+
+2. Full-Featured Thread Device with Interactive Command Line (`ot-cli-ftd`)
+
+     ```bash
+     $ cd ~/zephyrproject
+     $ rm -rf build_ot_cli_ftd
+     $ west build -b tlsr9518adk80d -d build_ot_cli_ftd zephyr/samples/net/openthread/cli -- -DOVERLAY_CONFIG=overlay-telink-fixed-mac.conf -DCONFIG_OPENTHREAD_FTD=y
+     ```
+
+### Firmware Flash
+
+Connect a TLSR9518 development board to the Burning Kit using a short blue USB cable as illustrated in the figure below.
+
+<img src="img/connection_overview.jpg" alt="connection_overview.jpg" width="624.00" />
+
+In the command line, execute the following commands to perform firmware burning (using the flashing of the `ot-cli-ftd` firmware as an example).
+
+```bash
+$ cd ~/zephyrproject/build_ot_cli_ftd/zephyr
+$ cp zephyr.bin ~/LinuxBDT/bin/ot-cli-ftd.bin
+$ cd ~/LinuxBDT
+$ sudo ./bdt 9518 ac
+ Activate OK!
+$ sudo ./bdt 9518 wf 0 -i bin/ot-cli-ftd.bin
+ EraseSectorsize...
+ Total Time: 2181 ms
+ Flash writing...
+ [100%][-] [##################################################]
+ File Download to Flash at address 0x000000: 491700 bytes
+ Total Time: 30087 ms
+```
+
+The flash method for `ot-rcp` is basically the same as that for `ot-cli-ftd`.
+However, please be aware that there are differences in the firmware paths and  names.
+
+After flashing, distinguish the two TLSR9518 development boards by marking them accordingly.
+Label the board flashed with `ot-cli-ftd` as "FTD Joiner" and the board flashed with `ot-rcp` as "RCP".
+
+## Configure a Serial Console for the FTD Joiner Device
+
+Duration: 3:00
+
+To control the FTD Joiner device through the CLI, connect a USB to TTL Serial Convertor Cable to the following pins on the TLSR9518 development board：
+
+| Name  | Pin                 |
+| :---- | ------------------- |
+| RX    | PB3 (pin 15 of J34) |
+| TX    | PB2 (pin 18 of J34) |
+| GND   | GND (pin 23 of J50) |
+
+> aside positive
+>
+> **Note:** Baud rate：115200 bits/s
+
+After connecting the Convertor Cable to the Linux machine, Open PuTTY.
+Then create a new terminal, set the serial port information, and open the serial port.
+
+<img src="img/uart_console.png" alt="uart_console.png" width="624.00" />
+
+ The OpenThread command line reference is here: [OpenThread CLI Reference](https://github.com/openthread/openthread/blob/f7690fe7e9d638341921808cba6a3e695ec0131e/src/cli/README.md).
+ Be sure to prefix with `ot` when using it.
```

Comment:
Be sure to prefix all commands with `ot`.

Reply:
Resolved in the updated commit.